libs/core/algorithms/include/hpx/parallel/algorithms/uninitialized_relocate.hpp

```diff
@@ -644,5 +700,87 @@ namespace hpx::experimental {
+            BiIter1 first, BiIter1 last,
+            BiIter2 dest_last) noexcept(util::detail::relocation_traits<BiIter1,
+            BiIter2>::is_noexcept_relocatable_v)
+        {
+            static_assert(hpx::traits::is_bidirectional_iterator_v<BiIter1> &&
+                "The 'first' and 'last' arguments must meet the requirements "
+                "of input iterators.");
+            static_assert(hpx::traits::is_bidirectional_iterator_v<BiIter2>,
+                "The 'dest_last' argument must meet the requirements of a "
+                "forward iterator.");
+            static_assert(util::detail::relocation_traits<BiIter1,
+                              BiIter2>::valid_relocation,
+                "Relocating from this source type to this destination type is "
+                "ill-formed");
+            // if count is representing a negative value, we do nothing
+            if (first == last)
+            {
+                return dest_last;
+            }
+
+            return parallel::util::get_second_element(
+                hpx::parallel::detail::uninitialized_relocate_backward<
+                    parallel::util::in_out_result<BiIter1, BiIter2>>()
+                    .call(hpx::execution::seq, first, last, dest_last));
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename BiIter1, typename BiIter2,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy_v<ExPolicy> &&
+                hpx::traits::is_iterator_v<BiIter1> &&
+                hpx::traits::is_iterator_v<BiIter2>
+            )>
+        // clang-format on
+        friend typename hpx::parallel::util::detail::algorithm_result<ExPolicy,
+            BiIter2>::type
+        tag_fallback_invoke(uninitialized_relocate_backward_t,
+            ExPolicy&& policy, BiIter1 first, BiIter1 last,
+            BiIter2 dest_last) noexcept(util::detail::relocation_traits<BiIter1,
+            BiIter2>::is_noexcept_relocatable_v)
+        {
+            static_assert(hpx::traits::is_input_iterator_v<BiIter1>,
+                "The 'first' and 'last' arguments must meet the requirements "
+                "of bidirectional iterators.");
+            static_assert(hpx::traits::is_forward_iterator_v<BiIter2>,
+                "The 'dest' argument must meet the requirements of a "
+                "bidirectional iterator.");
+            static_assert(util::detail::relocation_traits<BiIter1,
+                              BiIter2>::valid_relocation,
+                "Relocating from this source type to this destination type is "
+                "ill-formed");
+
+            auto count = std::distance(first, last);
+
+            // if count is representing a negative value, we do nothing
+            if (hpx::parallel::detail::is_negative(count))
+            {
+                return parallel::util::detail::algorithm_result<ExPolicy,
+                    BiIter2>::get(HPX_MOVE(dest_last));
+            }
```

Comment:
This would be "library UB" according to P1144. You're welcome to treat it as a no-op (documented or undocumented), but it might be more appropriate to assert-fail or something, I don't know.